src/main/java/com/rocketden/main/service/ProblemService.java

```diff
@@ -83,6 +86,71 @@ public ProblemDto getProblem(String problemId) {
         return ProblemMapper.toDto(problem);
     }
 
+    public ProblemDto editProblem(String problemId, ProblemDto updatedProblem) {
+        Problem problem = repository.findProblemByProblemId(problemId);
+
+        if (problem == null) {
+            throw new ApiException(ProblemError.NOT_FOUND);
+        }
+
+        if (updatedProblem.getName() == null || updatedProblem.getDescription() == null
```

Comment:
Not sure this is necessary for this PR, but I notice that we do a lot of these null checks, where we iterate over the different fields and check if they are null -- and if so, throwing an error. I had an idea that on a future PR, we could simplify this by making a single utility function that iterates over the different fields to check if all have a value, similar to the checkLocation function we did on the frontend. Thoughts?

Reply:
Also, just on this PR, could we check that `updatedProblem == null` in this condition as well? We do that for the `problem`, but not for `updatedProblem`, meaning the wrong error could be thrown.

Reply:
Yeah I def think that could be helpful in the future - also sounds good, I added that null check